### ISSUE TITLE
Loss becomes NaN when using custom data. Clarification on data requirements

### ISSUE THREAD

**Issue by 42elenz** [alexjamesing/Deep_LVPM#6]
Hi,

First of all, thank you for providing this awesome repository! It is really rare that a Tutorial of a git works so seamlessly.  The tutorials and overall structure are really helpful, and it was easy to run the code, especially the tutorial in the README.

I did notice that the tutorial in the dedicated tutorial folder is slightly different from the one in the README, and it seems to have a few minor errors. Just wanted to point that out in case it helps for future updates (Missing Layer definition between in and output).

I tried running the model with my own data, but I keep getting NaN values for the loss during training for some specifications. Could you clarify whether the input data needs to meet specific requirements (e.g., normalization, certain value ranges, etc.)?

**My setup:**

Two modalities
250 samples
Biomarker data: 5 columns  (one ordinal, other metric)

Freesurfer tables: 324 columns (metric)

No Missing values.

Do these dimensions sound compatible with your implementation? Or is there a recommended preprocessing step or property that the data must have?
When could a cross_metric of nan arise? (no match at all, perfect match?)
I managed to run the dataset with another subgroup, so it does not seem to be the datatype itself.

I already lowered the L1 and L2 regularization. 

Here is a snippet:
Epoch 1/50
7/7 ━━━━━━━━━━━━━━━━━━━━ 13s 14ms/step - cross_metric: nan - mse_loss: 0.0457 - total_loss: 0.8711   
Epoch 2/50
**7/7 ━━━━━━━━━━━━━━━━━━━━ 0s 14ms/step - cross_metric: nan - mse_loss: nan - total_loss: nan**
Epoch 3/50
7/7 ━━━━━━━━━━━━━━━━━━━━ 0s 16ms/step - cross_metric: nan - mse_loss: nan - total_loss: nan

I hope you can help me with that, and thanks again for this valuable research :)

**Comment by alexjamesing** [comment]
Hi Esra,
Thanks a lot for your interest and kind words! Ah yes, this version of the tutorial was written to be compatible with an older version of tensorflow. I also noticed these errors when i upgraded tensorflow, and amended the README. I had forgotten about the tutorial.py script. I will amend this today, thanks for pointing this out! 
It shoudnt be necessary to normalize the data. The only thing i can think of immediately is to make sure your batch size is large enough (you could try 125 which would split your data in half). Also, one of the feature sets is quite small (5 columns), ndims must be 4 or less in this case. In cases where these features are correlated, maybe less. I would start with ndims = 1 and see if the problem persists. If this does not work, and If you feel comfortable, you can send me your scripts and even data to me at: alexing1421 [at] gmail.com and i will take a look. Lastly, if you find the code useful, please remember to give it a star! That is the only way i can justify spending time expanding or maintaining it, which i am very keen to do!
Best Wishes
Alex

**Comment by 42elenz** [comment]
It looks like the batch size was indeed the main issue for training.
The final evaluation still yields NaN, but I have a test-set and I can also read the cross_metric as an output from the training.  

Epoch 49/50
2/2 ━━━━━━━━━━━━━━━━━━━━ 0s 37ms/step - cross_metric: 0.2435 - mse_loss: 0.0437 - total_loss: 36.1790
Epoch 50/50
2/2 ━━━━━━━━━━━━━━━━━━━━ 0s 44ms/step - **cross_metric: 0.2459** - mse_loss: 0.0436 - total_loss: 36.1768
7/7 ━━━━━━━━━━━━━━━━━━━━ 1s 6ms/step - cross_metric: nan - mse_loss: 4.1542 - total_loss: 81.3923     
The mean correlation between data-types connected by the path model is **r=nan**

But unfortunately also by just evaluating, I still get nan:
mean_corr_test = DLVPM_Structural_instance.evaluate(X_arr[:40])
7/7 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - **cross_metric: nan** - mse_loss: 4.1542 - total_loss: 81.3923 


For test it works:]  
mean_corr_test = DLVPM_Structural_instance.evaluate(X_arr_test)
print('The mean correlation between data-types connected by the path model is r=' + str(mean_corr_test[1]))
1/1 ━━━━━━━━━━━━━━━━━━━━ 0s 48ms/step - **cross_metric: 0.2162** - mse_loss: 3.9399 - total_loss: 79.0343

P.S.:
Repo got Starred 🥇

**Comment by 42elenz** [comment]
The problem was solved by changing the batchsize on evaluation to 125. 

mean_corr = DLVPM_Structural_instance.evaluate(X_arr, batch_size=125)

**Comment by 42elenz** [comment]
Thanks for the help!
